### ISSUE TITLE
"SecurityError: The operation is insecure" on Firefox

### ISSUE THREAD

**Issue by szepeviktor** [GoogleChromeLabs/pwa-wp#248]
Please check the returned Promise. It could be rejected with a certain cookie setting
See https://bugzilla.mozilla.org/show_bug.cgi?id=1429714#c1

![kép](https://user-images.githubusercontent.com/952007/75777928-2c993f00-5d57-11ea-959e-4656ff0e6d2b.png)

**Comment by westonruter** [comment]
So we're missing a `catch()` call here?

https://github.com/GoogleChromeLabs/pwa-wp/blob/261c15ee65f643b5dec4927d7ece403b8b02424f/wp-includes/service-workers.php#L161-L186

**Comment by szepeviktor** [comment]
Seems like...

**Comment by westonruter** [comment]
Would you like to open a PR? Should it be putting the error in the console via `console.error()`?

**Comment by szepeviktor** [comment]
Thank you for your offer! I better not touch frontend code.

**Comment by westonruter** [comment]
I can't reproduce this issue. In private browsing, `navigator.serviceWorker` is not even defined. The same goes if I select to not remember history in non-private browsing.

**Comment by szepeviktor** [comment]
Try telling Firefox to clear cookies on exit - that is mentioned in the bug tracker.
about:preferences#privacy

![kép](https://user-images.githubusercontent.com/952007/75850327-5437e800-5de7-11ea-8322-c12690681398.png)

**Comment by westonruter** [comment]
OK, I can see that now:

<img width="858" alt="Screen Shot 2020-03-04 at 21 14 34" src="https://user-images.githubusercontent.com/134745/75950259-88e68680-5e5d-11ea-935b-97987e90e0e7.png">

With this patch:

```diff
diff --git a/wp-includes/service-workers.php b/wp-includes/service-workers.php
index 2197787..17cb864 100644
--- a/wp-includes/service-workers.php
+++ b/wp-includes/service-workers.php
@@ -183,6 +183,9 @@ function wp_print_service_workers() {
 									} );
 								} );
 							<?php endif; ?>
+						} )
+						.catch( ( err ) => {
+							console.info( "Service worker will not be installed due to:", err );
 						} );
 
 						<?php if ( is_admin_bar_showing() && ! wp_service_worker_skip_waiting() ) : ?>
```

The result is:

<img width="859" alt="Screen Shot 2020-03-04 at 21 15 47" src="https://user-images.githubusercontent.com/134745/75950267-8dab3a80-5e5d-11ea-8326-d4da2753d780.png">

Doesn't seem a whole lot better.

**Comment by szepeviktor** [comment]
I think we need `.reject()`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject

**Comment by westonruter** [comment]
No, I don't think that is right. The `reject` is already being called today by the browser today. When a rejected promise is not `caught` then it shows up on the console as an error. You can see from my code above that adding `catch` does prevent the error from showing in the console, but the Firefox DevTools still shows another error regardless. So I don't know how much benefit there is to `catch` it in the first place.

**Comment by szepeviktor** [comment]
Yes, we have the same problem as others in the Mozilla bug:

> Neither try/catch nor then/catch seems to be able to supress the error warnings in the console.

https://bugzilla.mozilla.org/show_bug.cgi?id=1429714#c12
